zotonic/zotonic
# Add filewatcher support for symlinks & added sites/modules

### Description

Fix filewatch for symlinked priv/src/include directories.

To do:

 - [x] Test with fswatch
 - [x] Test with inotify
 - [x] Test with zotonic_filewatcher_monitor
 - [x] Add support for new apps in ~~`user/modules`, `user/sites`~~ and `_build/default/lib`

~~If a new app is found (in user/modules and/or user/sites):~~

 1. ~~Wait till no further updates during a couple of seconds~~
 2. **Manually** Run rebar3 (`z:compile()`)

If a new directory (application) is found in `_build/default/lib`:

 1. Wait till no further updates during a couple of seconds
 2. Ensure that the code load path includes the new app
 3. Restart the running file watcher so that it adds the new app
 4. Reindex the sites and modules (do not automatically start the new application)

In this way checking out a module or site in the user directory, manually compiling it will automatically add all the new code paths to the running Zotonic node.

### Checklist

- [x] no BC breaks
